import/index_java/src/org/vufind/index/ConfigManager.java

```diff
@@ -210,7 +210,17 @@ public Ini loadConfigFile(String filename)
                 retVal.put(key, overrideSection.get(key));
             }
         }
-        return retVal;
+
+        // Return a copy of the section.
+        // If there are other operations performed later (e.g. sanitize)
+        // we do not want them to update the cache, because they might cause
+        // problems when executing them multiple times, like
+        // e.g. in multithreaded scenarios.
+        Map<String, String> retValCopy = new ConcurrentHashMap<>();
```

Comment:
It might even get more complicated.

For sections, the sanitizer will only be active if you call getSanitizedConfigSection.
For settings, getConfigSetting will always sanitize the settings, so there is no way to get a non-sanitized setting right now (only by doing a workaround via getConfigSection).
=> If we have 2 versions of getConfigSection, shoulnd't we have a getSanitizedConfigSetting as well and sanitize only if this one is called?

For sections, there are 2 public methods: getConfigSection and getSanitizedConfigSection.
For settings, there is only 1 public method: getConfigSetting.
=> Since getConfigSetting does not use getConfigSection, we need to fix the logic in both code paths.

Reply:
Good point! I wish we could figure out a good way to write unit tests for this Java code, because these are all the kinds of things that would be more easily addressed with test coverage. But that's a whole other rabbit hole...

It does seem to make sense to add a getSanitizedConfigSetting method... and possibly to refactor the existing getConfigSetting to leverage getConfigSection so we don't have to implement the same logic in multiple places.